src/main/java/org/prebid/server/bidder/sharethrough/SharethroughBidder.java

```diff
@@ -1,244 +1,189 @@
-        final ExtData extData = extImpSharethrough.getData();
-        return StrUriParameters.builder()
-                .pkey(extImpSharethrough.getPkey())
-                .bidID(imp.getId())
-                .gpid(extData != null ? extData.getPbAdSlot() : null)
-                .consentRequired(isConsentRequired)
-                .consentString(consentString)
-                .usPrivacySignal(usPrivacy)
-                .instantPlayCapable(canBrowserAutoPlayVideo)
-                .iframe(extImpSharethrough.getIframe())
-                .height(size.getHeight())
-                .width(size.getWidth())
-                .theTradeDeskUserId(ttdUid)
-                .sharethroughUserId(buyeruid)
-                .body(body)
+    private BidRequest modifyRequest(Imp imp,
+                                     BidRequest bidRequest,
+                                     ExtImpSharethrough extImpSharethrough,
+                                     Price bidFloorPrice) {
+        return bidRequest.toBuilder()
+                .bcat(extImpSharethrough.getBcat())
+                .badv(extImpSharethrough.getBadv())
+                .imp(Collections.singletonList(modifyImp(imp, extImpSharethrough.getPkey(), bidFloorPrice)))
+                .source(updateSource(bidRequest.getSource()))
                 .build();
     }
 
-    /**
-     * Make Headers for request.
-     */
-    private MultiMap makeHeaders(Device device, String page) {
-        final MultiMap headers = HttpUtil.headers()
-                .add("Origin", requestUtil.getHost(page))
-                .add("Referer", page);
-        final String ip = device.getIp();
-        if (ip != null) {
-            headers.add("X-Forwarded-For", ip);
-        }
-
-        final String ua = device.getUa();
-        if (ua != null) {
-            headers.add("User-Agent", ua);
-        }
-        return headers;
+    private static Imp modifyImp(Imp imp, String pKey, Price bidFloorPrice) {
+        return imp.toBuilder()
+                .tagid(pKey)
+                .bidfloor(bidFloorPrice.getValue())
+                .bidfloorcur(bidFloorPrice.getCurrency())
+                .build();
     }
 
-    /**
-     * Make {@link HttpRequest} from uri and headers.
-     */
-    private HttpRequest<SharethroughRequestBody> makeHttpRequest(
-            MultiMap headers, Date date, StrUriParameters strUriParameter, List<BidderError> errors) {
-
-        final String uri;
-        try {
-            uri = SharethroughUriBuilderUtil.buildSharethroughUrl(
-                    endpointUrl, SUPPLY_ID, VERSION, DATE_FORMAT.format(date), strUriParameter);
-        } catch (IllegalArgumentException e) {
-            errors.add(BidderError.badInput(e.getMessage()));
-            return null;
-        }
+    private Source updateSource(Source source) {
+        return source != null
+                ? source.toBuilder().ext(createExtSource()).build()
+                : Source.builder().ext(createExtSource()).build();
```

Comment:
in go they are not just replacing `Source.ext` with new one. If `Source.ext` is present, it will be just updated with `str` and `version` fields.